src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,38 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
-import core.basesyntax.model.User;
+import core.basesyntax.service.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_AND_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Login: " + user.getLogin()
+                    + " is already taken! Choose another one");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
```

Comment:
According to the checklist, it's recommended to split one big if into several small ones. However, you should check for null login before checking if the login already exists in the storage to prevent unnecessary storage access and potential `NullPointerException`.